actors/miner/tests/miner_actor_test_precommit_batch.rs

```diff
@@ -0,0 +1,320 @@
+    let mut rt = h.new_runtime();
+
+    let precommit_epoch = period_offset + 1;
+    rt.set_epoch(precommit_epoch);
+    h.construct_and_verify(&mut rt);
+    let dl_info = h.deadline(&rt);
+
+    let sector_nos: Vec<SectorNumber> = (0..batch_size).map(|x| x as u64 + 100).collect();
+    let sector_no_as_uints: Vec<u64> = (0..batch_size).map(|x| x as u64 + 100).collect();
+    let sector_expiration =
+        dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period; // on deadline boundary but > 180 days
+    let prove_commit_epoch = precommit_epoch + rt.policy.pre_commit_challenge_delay + 1;
+    let deal_lifespan = sector_expiration - prove_commit_epoch;
+
+    let mut sectors = vec![SectorPreCommitInfo::default(); batch_size];
+    let mut conf = PreCommitBatchConfig {
+        sector_weights: vec![SectorWeights::default(); batch_size],
+        first_for_miner: true,
+    };
+    let mut deposits = vec![BigInt::default(); batch_size];
+
+    for i in 0..batch_size {
+        let mut deals = DealSpec::default();
+        if deal_specs.len() > i {
+            deals = deal_specs[i].clone();
+        }
+        sectors[i] = h.make_pre_commit_params(
+            sector_nos[i],
+            precommit_epoch - 1,
+            sector_expiration,
+            deals.ids,
+        );
+
+        let deal_space = deals.size + deals.verified_size;
+        let deal_weight = BigInt::from(deals.size) * BigInt::from(deal_lifespan);
+        let verified_deal_weight = BigInt::from(deals.verified_size) * BigInt::from(deal_lifespan);
+        conf.sector_weights[i] = SectorWeights {
+            deal_space,
+            deal_weight: deal_weight.clone(),
+            verified_deal_weight: verified_deal_weight.clone(),
+        };
+        let pwr_estimate = qa_power_for_weight(
+            h.sector_size,
+            sectors[i].expiration - precommit_epoch,
+            &deal_weight,
+            &verified_deal_weight,
+        );
+        deposits[i] = pre_commit_deposit_for_power(
+            &h.epoch_reward_smooth,
+            &h.epoch_qa_power_smooth,
+            &pwr_estimate,
+        );
+    }
+    let net_fee = aggregate_pre_commit_network_fee(batch_size as i64, &base_fee);
+    let total_deposit: BigInt = deposits.iter().sum();
+    let total_balance = net_fee + total_deposit.clone();
+    rt.set_balance(total_balance + balance_surplus);
+
+    if exit_code != ExitCode::OK {
+        expect_abort_contains_message(
+            exit_code,
+            error_str,
+            h.pre_commit_sector_batch(
+                &mut rt,
+                PreCommitSectorBatchParams { sectors },
+                &conf,
+                base_fee,
+            ),
+        );
+
+        // State untouched.
+        let st: State = rt.get_state();
+        assert!(st.pre_commit_deposits.is_zero());
+        let expirations = h.collect_precommit_expirations(&rt, &st);
+        assert!(expirations.is_empty());
```

Comment:
Add an `rt.reset()` here to clear out expectations from pre_commit_sector_batch that were not fullfilled